lib/src/test/java/com/knuddels/jtokkit/reference/Cl100kBaseTest.java

```diff
@@ -0,0 +1,234 @@
+            measurements.put((int) i, ((endTime - startTime) / iterations));
+        }
+        measurements.forEach((i, t) -> System.out.println(i + "\t" + t));
+    }
+
+    @Test
+    void testEdgeCaseRoundTrips() throws Exception {
+        var testStrings = List.of(
+                "\n",
+                " ",
+                "a : b",
+                "  a",
+                "\n \n ",
+                "\n \n",
+                "\n ",
+                "\n \n!",
+                "\n \n   ",
+                "\n  !",
+                "\n A",
+                "  \n\r  \r\n  \r \n  A\nA \n A",
+                ",\n\n",
+                " ***\n\n\n\n",
+
+                "   !",
+                "   A",
+                "   0",
+                "   *",
+
+                "   \n!",
+                "   \nA",
+                "   \n0",
+                "   \n*",
+
+                "   \n !",
+                "   \n A",
+                "   \n 0",
+                "   \n *",
+
+                "Many words map to one token, but some don't: indivisible.\n\nUnicode characters like emojis may be split into many tokens containing the underlying bytes: \uD83E\uDD1A\uD83C\uDFFE\n\nSequences of characters commonly found next to each other may be grouped together: 1234567890",
+                "I paid $123,456 to 9876543210 people!",
+                "Mixed script: 你好 world! 🌍",
+                "Unicode snowman: ☃️",
+                "I'm:  0\n",
+                "We'll meet at 3 o'clock.",
+                "Hello, world! It's a beautiful day...",
+                "In 2023, I'll be 25 years old.",
+                "Hello \n\n World  !",
+                " It's 2:30pm;\n\n\n\nlet's eat, sleep , and code!",
+                "'Thank God, here it is.' But when we took up the trunk...",
+                "user@example.com",
+                "this is a 'quoted' word",
+                "　　a",
+                "'ſ",
+                "'ſ𣶸𣄬Ƙ淚",
+                "😩\n",
+                "03½",
+                "* ע",
+                "مرحبا بالعالم! كيف حالك؟ 😎",
+                "\u0000\uD81C\uDFE1 a\u0000b-\u0000\u0000\u0000 \u0000",
+                "🌍 a",
+                "(𥧙h",
+                ",   𰉄",
+                "  󵨐)",
+                "ﮀ\n ",
+                "😐𪶫X",
+                "෫𞅄",
+                "𬕹\n  ",
+                " 😈b\n\uD844\uDDAE'ſ\uD84F\uDDB8\uD84C\uDD2CƘ淚",
+                "𗭾  󻥹\n\uD875\uDDB0蛇"
+        );
+
+        IntStream.range(0, testStrings.size()).forEachOrdered(i -> {
+            var testString = testStrings.get(i);
+            System.out.println("Validating `" + normalizeStringForTesting(testString) + "`");
+
+            var actualTokens = ENCODING.encode(testString);
+            var decoded = ENCODING.decode(actualTokens);
+            assertEquals(testString, decoded, decoded);
+        });
+    }
+
+    @Test
+    public void testRoundTripWithRandomStrings() throws Exception {
```

Comment:
generates random valid UTF8 strings and checks if decoding and reencoding results in the same value